include/gatekeeper_gk.h

```diff
@@ -81,7 +99,21 @@ struct gk_nexthop {
 	struct ipaddr ip_addr;
 
 	/* The cached Ethernet header. */
-	struct ether_cache eth_cache;
+	struct ether_cache *eth_cache;
+};
+
+struct neighbor_hash_table {
+	/* The prefix length, for which the hash table is responsible. */
+	int prefix_len;
```

Comment:
Given that we'll have a single neighbor FIB entry per interface per IP version, once the table is created, we don't need to know the prefix length. So we can drop the field `prefix_len`.